feat: ability to update personal profile picture

Sboonny opened the pull request:
<!-- Please follow the below checklist and put an `x` in each of the boxes to agree with the statement, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

<!-- If your pull request closes a GitHub issue, replace the XXXXX below with the issue number. For e.g. Closes #12. The issue #12 will automatically get closed when this PR gets merged. -->

this aims to create, profile picture field which will be used by `Avatar` chakra component to display personal characteristics into profile picture. 

Will I attempt to block any type of images? nope, just adding the ability to upload an image, more than can be dealt with post MVP


the outcome 

![image](https://user-images.githubusercontent.com/88248797/193507493-d3863b23-d986-475d-8060-b7b4255e1202.png)

Commit:
add user_image to database

Commit:
apollo hook can fill image_url empty fields

Commit:
Add image_url field to profileform

Commit:
add user.image to avatar, try to change image in the form

Commit:
send image_url data to the server

gikf commented on client/src/components/PageLayout/Header.tsx:
```diff
@@ -78,6 +78,7 @@ export const Header: React.FC = () => {
 
   const goHome = () => router.push('/');
 
+  console.log(user?.image_url);
```
```suggestion
```

gikf reviewed (commented):
I don't know if that's intended or not. When using transparent image (ie. https://cdn-icons-png.flaticon.com/512/149/149157.png), the background color that's set by the user name will still be used.

![image](https://user-images.githubusercontent.com/60067306/193640662-b77ebad3-fd2e-43de-8e54-3c5ad45a2695.png) ![image](https://user-images.githubusercontent.com/60067306/193640804-9f638864-8647-4447-b3ba-384e84e67c31.png)

Sboonny committed:
Remove console log, and format space

Co-authored-by: Krzysztof G. <60067306+gikf@users.noreply.github.com>

Sboonny commented:
It isn't intended, it's chakra doing good work.

That being said, do we want transparent image? I think it's harmful for UX, but we don't have users yet so we can have it 🤔 and test it

Sboonny commented:
Stop it as 100%? I don't think so. 

But we can create an `::after` content (maybe a nice trending glass effect) that stay behind the image, if the image is transparent, then that `::after` content will show up instead (well technically the image will disappear and what behind it (::after) will show up).

Should I give that a go and try to override chakra background color?

ojeytonwilliams commented:
I could be underthinking it, but how about setting the background to white if there's an image?

Sboonny commented:
There are pages that use `avatar` component in them, mainly ChapterPage and EventPage, although this isn't affecting them yet. it should be DRYed and used on these pages. if we're going to DRY it, using background white won't work in UX aspect but the idea is on the mark 👍

Sboonny commented:
I have reread this, and I was confused. I though we didn't want a transparent Image, but we do want transparent photo, I am adding them 👍

Commit:
Allow transparent Images

ojeytonwilliams committed:
Merge branch 'main' into add/user-picture

ojeytonwilliams commented on server/prisma/schema.prisma:
```diff
@@ -264,6 +264,7 @@ model users {
   updated_at       DateTime        @updatedAt
   id               Int             @id @default(autoincrement())
   name             String
+  image_url        String
```
```suggestion
  image_url        String?
```
Having this be nullable better reflects the fact that the image_url is optional. It also means that `image_url: ''` can be deleted everywhere.

ojeytonwilliams reviewed (changes requested):
I thi

Sboonny committed:
Make image_url nullable in the schema

Co-authored-by: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

Commit:
changed the queires and input fields to account for null

Commit:
Merge branch 'main' into add/user-picture

Commit:
Merge branch 'freeCodeCamp:main' into add/user-picture

ojeytonwilliams commented on server/prisma/generator/factories/user.factory.ts:
```diff
@@ -19,27 +19,31 @@ const createUsers = async (
     email: 'foo@bar.com',
     name: 'The Owner',
     instance_role: { connect: { id: instanceRoles.owner.id } },
+    image_url: '',
```
```suggestion
```

Sboonny replied:
Oh 😅, they were an attempt to please typescript, is there a better way around that type error?🤔

ojeytonwilliams commented on server/prisma/generator/factories/user.factory.ts:
```diff
@@ -19,27 +19,31 @@ const createUsers = async (
     email: 'admin@of.chapter.one',
     name: 'Chapter One Admin',
     instance_role: { connect: { id: instanceRoles.chapter_administrator.id } },
+    image_url: '',
```
```suggestion
```

ojeytonwilliams commented on server/prisma/generator/factories/user.factory.ts:
```diff
@@ -19,27 +19,31 @@ const createUsers = async (
     email: 'admin@of.chapter.two',
     name: 'Chapter Two Admin',
     instance_role: { connect: { id: instanceRoles.chapter_administrator.id } },
+    image_url: '',
```
```suggestion
```

ojeytonwilliams commented on server/prisma/generator/factories/user.factory.ts:
```diff
@@ -19,27 +19,31 @@ const createUsers = async (
     email: 'banned@chapter.admin',
     name: 'Banned Chapter Admin',
     instance_role: { connect: { id: instanceRoles.chapter_administrator.id } },
+    image_url: '',
```
```suggestion
```

ojeytonwilliams commented on server/prisma/generator/factories/user.factory.ts:
```diff
@@ -48,6 +52,7 @@ const createUsers = async (
     email: 'test@user.org',
     name: 'Test User',
     instance_role: { connect: { id: instanceRoles.member.id } },
+    image_url: '',
```
```suggestion
```

ojeytonwilliams commented on server/prisma/generator/factories/user.factory.ts:
```diff
@@ -58,6 +63,7 @@ const createUsers = async (
       email: internet.email(),
       name: `${name.firstName()} ${name.lastName()}`,
       instance_role: { connect: { id: instanceRoles.member.id } },
+      image_url: '',
```
```suggestion
```

ojeytonwilliams commented on server/src/app.ts:
```diff
@@ -120,6 +120,7 @@ export const main = async (app: Express) => {
             name: 'member',
           },
         },
+        image_url: '',
```
```suggestion
```

ojeytonwilliams commented on server/tests/fixtures/users.ts:
```diff
@@ -38,6 +38,7 @@ const baseUser: User = {
   created_at: new Date(),
   updated_at: new Date(),
   email: 'an@add.ress',
+  image_url: '',
```
```suggestion
```

ojeytonwilliams reviewed (changes requested):
Pretty sure we don't need any of these.

Sboonny committed:
Remove image_url: '', in users input

Co-authored-by: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

ojeytonwilliams committed:
fix(test): set image_url to be null in fixture

It is optional (i.e. can be undefined) when part of the input used to
create an entry in the db, but it the return type is string | null

ojeytonwilliams commented:
@Sboonny I seem to have upset Cypress. Sorry about that!  I'll fix it shortly.

ojeytonwilliams committed:
Merge branch 'main' into add/user-picture

Sboonny commented:
no worries, feel free to 👍

ojeytonwilliams committed:
fix: allow image_url to be nullable

ojeytonwilliams reviewed (approved):
LGTM :+1: Assuming the CI passes, :crossed_fingers:, this seems ready to me.
